### DISCUSSION TITLE
API to insert record [Jul 01]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#2617] [Discord Threads]
> 🙋‍♂️ **Trung BUI**

**Comment by o1lab** [comment]
> 👨‍🦲 **dstala**  
#### <@899573201598836747> You can use API to insert record   
#### https://apis.nocodb.com/#tag/DB-table-row/operation/db-table-row-create  
#  
> 🙋‍♂️ **Trung BUI**  
#### Do you know how can i get REST API Schema ?  
#  
> 💁 **o1lab**  
#### you would have to simply go to respective project's swagger schema  
#  
> 💁 **o1lab**  
#### bottom left  
#  
> 🙋‍♂️ **Trung BUI**  
#### I mean are there any REST APIs to get the schema or not ?  
#  
> 🙋‍♂️ **Trung BUI**  
#### I know i can find in swagger schema but i want get schema from api in python actually.  
#  
> 👨‍🦲 **dstala**  
#### Get list of projects & to derive project-id   
#### GET /api/v1/db/meta/projects   
####    
#### Get list of tables in a project & to derive table-id   
#### GET /api/v1/db/meta/projects/{projectId}/tables   
####    
#### Get table schema using table-id   
#### GET /api/v1/db/meta/tables/{tableId}  
#  
> 👨‍🦲 **dstala**  
#### Code snippet example for python can be found by clicking `code snippet` option once you open any table in UI at right bottom corner   
####    
#### ```   
#### import http.client   
####    
#### conn = http.client.HTTPConnection("localhost:8080")   
####    
#### headers = { 'xc-auth': "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6InVzZXJAbm9jb2RiLmNvbSIsImZpcnN0bmFtZSI6bnVsbCwibGFzdG5hbWUiOm51bGwsImlkIjoidXNfMGZnY3cyNnY4bDMzY2YiLCJyb2xlcyI6InVzZXIsc3VwZXIiLCJ0b2tlbl92ZXJzaW9uIjoiOTQ0MWExMGY3MTk1ZWI0ZThiNTViYzdkOWVmOTBmY2RlMjViZjM2NGQ2MTkwZWIzZGMxOWI5M2I5NDVmNjAwNTA1MzM3MWExYTI3ZGQ5NTAiLCJpYXQiOjE2NTY0OTY5MTMsImV4cCI6MTY1NjUzMjkxM30.uuzJdgVXYjBSzzADFSxTGjqfO3YZawnm-E1CIAlbZMg" }   
####    
#### conn.request("GET", "/api/v1/db/data/noco/p_wiukyreqe052rf/Actor/views/Grid%20view?limit=25&offset=0&where=", headers=headers)   
####    
#### res = conn.getresponse()   
#### data = res.read()   
####    
#### print(data.decode("utf-8"))   
#### ```  
#  
> 👨‍🦲 **dstala**  
#### You need to configure auth token appropriately (right top 3-dot menu, `Copy Auth Token`)  
#  
> 🙋‍♂️ **Trung BUI**  
#### Thanks a lot ! Can u help me how to use signup with API ? I can not to use it.  
![image0](https://cdn.discordapp.com/attachments/991357573712314408/991712259195817994/unknown.png)  
#  
> 🙋‍♂️ **Trung BUI**  
#### With404 error!  
#  
> 🙋‍♂️ **Trung BUI**  
#### How can i create users automatically from script in python ? Is that possible ?  
#  
> 👨‍🦲 **dstala**  
#### did we ensure we passed appropriate credentials for xc-auth?  
#  
> 👨‍🦲 **dstala**  
#### refer API details here (projectUserList/Add/Update/Remove)   
#### https://docs.nocodb.com/developer-resources/rest-apis#meta-apis  
#  
> 🙋‍♂️ **Trung BUI**  
#### I'm trying login by api from my app, but when i was connecting dashboard, it always ask me to login.  
#  
> 🙋‍♂️ **Trung BUI**  
#### Someone can help me 😭  
#  
> 🙋‍♂️ **Trung BUI**  
#### My app is running on Django at port 8081 and nocodb at port 8080, i'm trying to login the user when they click on a button form my app  
#  
> 💁 **o1lab**  
#### sorry, its not clear if you are login via API and access dashboard - you would still have to login via our UI to access dashboard 😅  
#  
> 🙋‍♂️ **Trung BUI**  
#### Yes, it's problem, so how i can access directly to dashboard by trying call API login  
#  
> 💁 **o1lab**  
#### if I understand correctly, there is no way for noco to know that you have logged in via a different port  
#  
> 💁 **o1lab**  
#### Im trying to say - that is not the right way to go about it  
#  
> 💁 **o1lab**  
#### if you want to embed dashboard into your app   
#### you can share the entire base and you can embed that as an iframe  
#  
> 🙋‍♂️ **Trung BUI**  
#### Sorry for my english, so, i was created a button from my app and i have already a list of user, now, i want when users click on that button, i call api login then they can access directly to dashboard without login with your UI dashboard, it's possible ?  
#  
> 💁 **o1lab**  
#### it wont be possible - its incorrect way to go about things unless Im not understanding your point  
#  
> 💁 **o1lab**  
#### you can access the data via our APIs after login    
#### but you cant access the dashboard - its on a different port  
#  
> 🙋‍♂️ **Trung BUI**  
#### yeah, sorry, i'm trying use HttpRedirect after login success to access your dashboard.  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/991357573712314408)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
